statchatui/src/Compare/Compare.jsx

```diff
@@ -43,34 +37,45 @@ const Compare = ({ signedInUserData }) => {
     setIsLoading(true); //sets the state to true before the api call is displaying the stats
     let stats1;
     let stats2;
+    let validUser = true;
     try {
       stats1 = await fetchStats(user1);
       stats2 = await fetchStats(user2);
     } catch (error) {
       setError(error?.message);
+      validUser = false;
     }
+    //if there was an error fetching the stats( meaning that the user does not exist) then it will not be added to the user queue
+    if (validUser) {
+      setOtherUserStats(stats1);
+      setSignedInUserStats(stats2);
+      setOtherUserFormData(user1);
+      setSignedInUserFormData(user2);
 
-    setOtherUserStats(stats1);
-    setSignedInUserStats(stats2);
-    setOtherUserFormData(user1);
-    setSignedInUserFormData(user2);
+      //checks if the user is in the userqueue already
+      const isUserInQueue = userQueue.some(
```

Comment:
Nice use of `.some`! What's the difference between `some` and using something like a for-loop?

Reply:
some returns true if even one value meets the condition while a for loop is for more general iterating things